js/payone/core/payolution.js

```diff
@@ -157,4 +157,23 @@ function switchInstallmentPlan(sKey, sCode, iInstallments)
     ];
     switchVisibility(aShow, true);
     $(sCode + '_selected_installmentplan').value = iInstallments;
+}
+
+/**
+ *
+ * @param payment_code
+ */
+function payonePayolutionCustomerDobInput(payment_code)
+{
+    var daySelect = $(payment_code + '_additional_fields_customer_dob_day');
+    var monthSelect = $(payment_code + '_additional_fields_customer_dob_month');
+    var yearSelect = $(payment_code + '_additional_fields_customer_dob_year');
+    var hiddenDobFull = $(payment_code + '_additional_fields_customer_dob_full');
+
+    if (daySelect == undefined || monthSelect == undefined || yearSelect == undefined
+        || hiddenDobFull == undefined)  {
+        return;
+    }
+
+    hiddenDobFull.value = yearSelect.value + "-" + monthSelect.value + "-" + daySelect.value;
```

Comment:
The jQuery object provides the .val() method to retrieve the value. Nevertheless .value property should work too. Maybe an additional check for valid values could make sense.

Reply:
The function is perfectly fine. The issue is simply that the files ratepay.js or debitpayment.js will not be loaded when the belonging payment methods are not active. Hence, the function is undefined. (In the concrete scenario we only use payolution without ratepay and directdebit methods.)